### ISSUE TITLE
Windows Explorer crash on Windows 7 x64

### ISSUE THREAD

**Issue by 60fpshacksrock** [GerbilSoft/rom-properties#286]
Problem is happening with version 1.7.3

The crash happens when say, trying to open a psx bin file with something such as pcsxr.
Though it's not related. Looking at the file in properties>ROM Properties or hovering 
over it with the mouse causes no problem either. It happens when a program uses a
different windows explorer dialog menu. 

It crashes when you try to scroll down, say on the desktop for instance.

I get a "COM Surrogate has stopped working".
Here is the problem signature if it helps:

Problem Event Name:	BEX
  Application Name:	DllHost.exe
  Application Version:	6.1.7600.16385
  Application Timestamp:	4a5bc6b7
  Fault Module Name:	StackHash_0a9e
  Fault Module Version:	0.0.0.0
  Fault Module Timestamp:	00000000
  Exception Offset:	00a7ba78
  Exception Code:	c0000005
  Exception Data:	00000008
  OS Version:	6.1.7601.2.1.0.256.1
  Locale ID:	1033
  Additional Information 1:	0a9e
  Additional Information 2:	0a9e372d3b4ad19135b953a78882e789
  Additional Information 3:	0a9e
  Additional Information 4:	0a9e372d3b4ad19135b953a78882e789

**Comment by GerbilSoft** [comment]
v1.7.2 fixed a crash bug when parsing some PlayStation disc images. What version of rom-properties do you have set up?

If you're using v1.7.3 or a nightly build, can you narrow down which specific PlayStation disc image is causing the crash?

**Comment by 60fpshacksrock** [comment]
As far as I can remember, any game. As said above, it does not crash with the normal explorer mode.
I have since started to use version 1.4.1.  Though now, after re-installing 1.7.3 it wont show any info
for PSX disc images 
![game](https://user-images.githubusercontent.com/59340087/111099517-0e465180-851c-11eb-94dd-1478f398055b.png)

and causes pcsxr's ISO open dialog to just exit. I may have crash reporting turned off on Windows 7 now
but it does not matter as far as I know.
It also does not have any ROM Properties info when right-clicking properties in Windows 7.![game2](https://user-images.githubusercontent.com/59340087/111100012-25397380-851d-11eb-9fb8-7aef4617b955.PNG) This new issue though may be with something I have installed call icaros, which gives thumbnails to media files that Windows
would not. I have also dumped this game several times before to make sure its MD5 checksum (A8CE602F3A846ADBBFAE7770CD33AFAB) matches the redump.

Also I should probably note that I have an unofficial Windows XP Luna theme running but I don't think it 
should causes any issues.

**Comment by SaraJLeen** [comment]
I've had issues with crashing on PS1 bins and some other types of ISOs/CSOs in rom-properties as well, on Windows 10 x64. I also have Icaros installed but that seems to be the only overlap. I ultimately worked around it by disabling rom-properties' support for those files on my end, but I definitely would rather I didn't have to.

**Comment by GerbilSoft** [comment]
Please test the latest nightly build and see if it crashes on those disc images:
https://ci.appveyor.com/project/GerbilSoft/rom-properties

Extract the x64 release package, open an admin command prompt, cd to the directory, and run `regsvr32 rom-properties.dll`.

If it works, then I'll look into releasing v1.8 this weekend.

**Comment by 60fpshacksrock** [comment]
Thank you.

It works. But I've had explorer restart twice.

By the way this tool is very neat. It sort of reminds me of (even though I don't use it unless I have to) Linux distributions as they tell certain ROM types. But this tool is even better and more advanced as it gives more internal file information as well as thumbnails for certain files that have banner or icon data such as GameCube and NDS respectively among others.

Also I was wondering (not asking) if you had maybe eventually thought about adding PS3 PARAM.SFO support.

**Comment by SaraJLeen** [comment]
I haven't had any issues since trying the nightly build.

**Comment by cheatfreak47** [comment]
I was just having crashes with 1.7.3 on the release build but since switching to the latest nightly, no longer getting crashes.
my crashes were in pcsx2's file picker for the record

**Comment by 60fpshacksrock** [comment]
Do the nightly builds have to be compiled or is there a link to pre-compiled builds?

**Comment by GerbilSoft** [comment]
Nightly builds are available here: https://ci.appveyor.com/project/GerbilSoft/rom-properties

Note that they're not packaged the same way as the release builds. Among other things, 32-bit and 64-bit are built separately. (Use the MSVC 2015 release builds.)

**Comment by 60fpshacksrock** [comment]
Thank for the link. 

My apologies for not scrolling back up the page to see the link you gave me before. Now I feel silly.

**Comment by NotaInutilis** [comment]
Hi, I've got a crash that might be related with PSP .cso files. I'm on Windows 10 x64 with 1.8.1 and as soon as I open the PSP folder full of CSO files, the explorer window crashes.

**Comment by GerbilSoft** [comment]
Can you narrow down which specific CSO file is crashing? (You may need to use cmd to move files between folders.)

**Comment by NotaInutilis** [comment]
After some quick testing, I can narrow it down to: all of them. I've converted them all with maxcso on default settings.

**Comment by GerbilSoft** [comment]
Obviously that's not good. I'll test this out later today and see what's causing the problem.

**Comment by GerbilSoft** [comment]
```
$ rpcli "300 - March to Glory (USA).cso"
== Reading file '300 - March to Glory (USA).cso'...
Floating point exception
```

Well, that's certainly something. (Division by zero!)

**Comment by GerbilSoft** [comment]
Found it. While reworking how zlib was detected, I accidentally added a `break;` statement here:

https://github.com/GerbilSoft/rom-properties/blob/master/src/libromdata/disc/CisoPspReader.cpp#L246

CISO and ZISO formats are very similar, except ZISO uses LZ4, while CISO usually uses zlib. (CISOv2 might use LZ4.) Hence, I had both formats using the same header initialization code. The extra `break;` prevents the header from being initialized for CISO (CSO) files, which results in the block size remaining as 0.

I'll fix this and release v1.8.3 for all platforms later tonight.

**Comment by NotaInutilis** [comment]
Good job! Thanks!

**Comment by GerbilSoft** [comment]
One thing that I definitely need to do for v1.9.0 is add a test suite with various file formats other than textures/icons/etc. This would include e.g. sample ISO images compressed with CSO, ZSO, and JSO, and verifying that they can be read correctly. If I had this test suite in v1.8.0, this error would have been caught much earlier, since the tests would crash.

**Comment by GerbilSoft** [comment]
Fixed in v1.8.3.

**Comment by 60fpshacksrock** [comment]
@GerbilSoft 
A new issue has arisen. 

Now it is crashing explorer simply by clicking on PSP ISO's. I even updated to the latest version and I am getting the same thing.
